test/unit/task_test.rb

```diff
@@ -21,6 +21,19 @@ class TasksTest < ActiveSupport::TestCase
     test 'can search the tasks by current_user in combination with implicit search' do
       assert_equal [@task_one], ForemanTasks::Task.search_for("owner.id = current_user AND #{@task_one.label}")
     end
+
+    test 'can search the tasks by user' do
```

Comment:
Since we have a specific brach in the logic for the `id`, it would be good to add some test around searching based on id. No other comments: tested and it worked well.

Reply:
Added a test for that